### PR TITLE
Download a file

### PR DESCRIPTION
**Pull request by plocket**
Merge after previous PR (#18) as this uses it.

### CODE REVIEW

**Comment by plocket** [comment]
Added downloads to .gitignore in files to push

File: lib/steps.js [hunk 1]
```diff
@@ -65,6 +73,40 @@ Given(/I start the interview at "([^"]+)"[ on ]?(.*)/, async (file_name, optiona
...
     scope.page.setDefaultTimeout(120 * 1000)
+    // Files should get downloaded
+    scope.docs_dir = `downloads_${ scope.scenario_name }`
+    await scope.page._client.send('Page.setDownloadBehavior', {
+      behavior: 'allow',
+      downloadPath: path.resolve( scope.docs_dir ),  // Save in root of this project
+    });
+
+    // ---- EVENT LISTENERS ----
+    // From https://stackoverflow.com/a/51423688
```
Comment (niharikasingh) [inline lib/steps.js:84]: This block seems like an incantation that I don't entirely understand, but if you've tested and it works, then I'm ready to believe in this spell 🧙

File: lib/scope.js [hunk 2]
```diff
@@ -121,4 +121,28 @@ module.exports = {
 
     return elem;
   },  // Ends getSpecifiedChoice
+
+  detectDownloadComplete: async function detectDownloadComplete(scope, endTime) {
+    /* Resolve if a download flag has changed. Timeout after a bit less than
+    *    the gobal permitted timeout */
+
+    // Stop/resolve if too much time has passed
+    let expirationTime = 1.9 * 60 * 1000;  // m * s * ms
```
Comment (niharikasingh) [inline lib/scope.js:130]: Not urgent, but I wonder if we should be using a global variable for the permitted timeout, and then this should just be `env_vars.TIMEOUT - 2000` or something.

**Review by niharikasingh — APPROVED** [review]

File: lib/scope.js [hunk 3]
```diff
@@ -121,4 +121,28 @@ module.exports = {
 
     return elem;
   },  // Ends getSpecifiedChoice
+
+  detectDownloadComplete: async function detectDownloadComplete(scope, endTime) {
+    /* Resolve if a download flag has changed. Timeout after a bit less than
+    *    the gobal permitted timeout */
```
[the remark this answers is not in the document]
Comment (plocket) [reply lib/scope.js:127]: gobal is such a great word, though!

File: lib/scope.js (hunk 2 above)
Comment (plocket) [reply lib/scope.js:130]: I really like that. I'll make it an issue.

File: lib/steps.js [hunk 4]
```diff
@@ -274,26 +316,47 @@ Then(/the "([^"]+)" link opens a working page/, async (linkText) => {  // √
...
 
   // Making this a race would probably be more simple
   let elem;
   if (elemType === 'button') {
-    [elem] = await scope.page.$x(`//button/span[contains(text(), "${phrase}")]`);
-    if ( !elem ) {
-      [elem] = await scope.page.$x(`//button[contains(text(), "${phrase}")]`);
+    // [elem] = await scope.page.$x(`//button/span[contains(text(), "${phrase}")]`);
+    // if ( !elem ) {
+      [elem] = await scope.page.$x(`//button//text()[contains(normalize-space(), "${phrase}")]/..`);
```
[the remark this answers is not in the document]
Comment (plocket) [reply lib/steps.js:351]: I think I take care of this in another PR, so I'm not going to mess with it right now in github.

File: lib/steps.js (hunk 1 above)
Comment (plocket) [reply lib/steps.js:84]: The sorcery of stackoverflow...

**Pull request merged**